.github/workflows/build.yml

```diff
@@ -331,33 +360,48 @@ jobs:
 
       - name: Update CI build info
         run: |
-          ./Scripts/update_app_ci_build_info.sh ${{ github.run_id }} ${{ github.run_number }} ${{ github.run_attempt }} "${{ inputs.compiler-flags }}"
+          ./Scripts/update_app_ci_build_info.sh ${{ github.run_id }} ${{ github.run_number }} ${{ github.run_attempt }} "${{ env.COMPILER_FLAGS }}"
 
       - name: Build iOS app
         run: |
-          ./Scripts/build.sh
+          ./Scripts/build.sh ${{ env.BUILD_MODE }}
 
-      - name: Prepare IPA & dSYM files for upload to GitHub
+      - name: Prepare artifacts for upload to GitHub
         run: |
-          mkdir -p export/dSYMs
-          cp build/Bitwarden/Bitwarden.ipa export
-          cp -rv build/Bitwarden.xcarchive/dSYMs/*.dSYM export/dSYMs
-
-      - name: Upload IPA & dSYM files
+          mkdir -p export
+          case "$BUILD_MODE" in
+            "Simulator")
+              cp -r build/DerivedData/Build/Products/Debug-iphonesimulator/Bitwarden.app export
+              #cp -r build/DerivedData/Build/Products/Debug-watchsimulator/BitwardenWatchApp.app export
+              ;;
+            "Device")
+              mkdir -p export/dSYMs
+              cp build/Bitwarden/Bitwarden.ipa export
+              cp -rv build/Bitwarden.xcarchive/dSYMs/*.dSYM export/dSYMs
+              ;;
+            *)
+              echo "Error: Invalid BUILD_MODE '$BUILD_MODE'"
+              exit 1
+              ;;
+          esac
+
+      - name: Upload artifacts to GitHub
         uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
         with:
-          name: Bitwarden iOS ${{ steps.version_info.outputs.version_name }} (${{ steps.version_info.outputs.version_number }}) ${{ env.BUILD_VARIANT }} ${{ env.XCODE_VERSION || env.DEFAULT_XCODE_VERSION }}
+          name: Bitwarden iOS ${{ steps.version_info.outputs.version_name }} (${{ steps.version_info.outputs.version_number }}) ${{ env.BUILD_VARIANT }} ${{ env.XCODE_VERSION || env.DEFAULT_XCODE_VERSION }} ${{ env.COMPILER_FLAGS }}
```

Comment:
I expected it to be trimmed but who knows, waiting for CI to test. Added it because some I had unrelated artifacts upload issues and it hit me that this could be an issue if we need two builds that only change in flags for some odd reason.

Reply:
That made me realise we also need to add something for build mode.